crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -661,6 +666,14 @@ where
             }
             ast::Stmt::Import(node) => {
                 for alias in &node.names {
+                    // Mark the imported module, and all of its parents, as being imported in this
+                    // file.
+                    if let Some(module_name) = ModuleName::new(alias.name.as_str()) {
+                        for name in module_name.parents() {
+                            self.imported_modules.insert(ModuleName::new(name).unwrap());
+                        }
```

Comment:
Nit:

* The usage suggest that we should change `parents` to return a `ModuleName`
* We can use `extend`
```suggestion
                        self.import_modules.extend(module_name.parents().map(|name| ModuleName::new(name).unwrap()))
```

Reply:
Done